src/processor/map/map_set.cpp

```diff
@@ -85,7 +86,12 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapSetNodeProperty(LogicalOperator
     for (auto& info : set->getInfos()) {
         executors.push_back(getNodeSetExecutor(info, *inSchema));
     }
-    auto printInfo = std::make_unique<OPPrintInfo>(set->getExpressionsForPrinting());
+    auto& info = set->getInfos();
+    std::string result = ExpressionUtil::toString(info[0].setItem);
```

Comment:
The problem with this is that info has a private type, only accessible through certain files. So I had to convert it to string in the map_set.cpp before I could bring it into the set.h and set.cpp files

Reply:
Which part do you need that's private? You can send infos over to PrintInfo and would still be able to access setItem. Or, you can create a vector of expression_pair similar to the for loop you have right now and send that. It is better to do the actual string conversions in toString.

Reply:
sending info through printInfo requires the other side to have public access to the     std::vector<std::unique_ptr<RelSetExecutor>> type. However since it is private in set.h, I cannot access it. That's why I convert it first in map_set.cpp and then send it over

Reply:
Oh yes I see the issue. 
https://github.com/kuzudb/kuzu/blob/433c8d303a1face982a62d4f69c0a51b5b6879e5/src/processor/map/map_index_scan_node.cpp#L27-L31

Try doing something similar to what I do in `map_index_scan_node` where you make your own vector with what you need and send that and do the string conversions in toString. That way you avoid the issue with `BoundSetPropertyInfo`

Reply:
Ok perfect, I will do that this weekend and hopefully have it done by monday